pkg/venafi/firefly/firefly.go

```diff
@@ -43,6 +43,10 @@ const (
 	scopesSeparator = " "
 )
 
+var (
+	rsaSizes = map[int]bool{certificate.DefaultRSAlength: true, 3072: true, 4096: true}
```

Comment:
Cant this be achieved with a simple array. Values are finite and array is small (4 or 5 values max). Iterating over such array is not expensive an easier to read and print on the error message

Reply:
I've implemented a kind of set as it's explained in the official documentation https://go.dev/doc/effective_go#maps

Reply:
I get it. However given the limited values on the array and the small length, this could be written just by using a simple array and comparing values. 

In the end you are not saving any performance or loop because you are already looping the entire map in order to build the values on the error message. I think it would be easier to read just to iterate the values array to find the size and jus print the array in the message as is. Its one loop in both approaches but:

- By using a map, you always need to loop the maximum number of iterations to build the error message.
- It is harder to read and understand what is happening.

Reply:
The iteration you mention only will happen when the passed rsa-size is not a valid one in order to be used for the error to thrown, when with the approach you mention the iteration will occur no matter if the rsa-size value is valid or not.